hal/network/ncp/cellular/network_config_db.h

```diff
@@ -21,6 +21,15 @@
 
 namespace particle {
 
+enum class CellularNetworkProv {
+    NONE = 0,
+    TELEFONICA = 1,
+    KORE_VODAFONE = 2,
+    KORE_ATT = 3,
+    TWILIO = 4,
+    MAX = 100,
```

Comment:
Is there a reason to add `100` for MAX? Should this be left to enumerate by itself, or set it to 5?

For example, if for some reason the `net_prov_` is '8' or some garbage number, `CellularNetworkConfig::hasNetProv()` will still return `true` while it's supposed to be false.

Reply:
I just set it high for when we add more, we don't have to remember to bump the MAX up.  The check for < MAX is kind of just a loose sanity check that we don't have some random garbage.  Even if there was a garbage value of 8, it wouldn't match anything in the enum list, so it shouldn't harm the code that uses this.  That said, I'll let it enumerate itself... that slipped my mind as an option when typing in the list.

Reply:
cool 👍

Reply:
I just removed MAX because it's an enum class vs. enum which I had started with.